### ISSUE TITLE
[4.0.0-beta.12] Ionic 4 - Angular 6: How to control Ionic router history to stop cache a view component?

### ISSUE THREAD

**Issue by Moelgaard85** [ionic-team/ionic#16472]
Ionic info
```
Ionic:

   ionic (Ionic CLI)             : 4.4.0 (C:\Users\Stationaer\AppData\Roaming\npm\node_modules\ionic)
   Ionic Framework               : @ionic/angular 4.0.0-beta.12
   @angular-devkit/build-angular : 0.7.4
   @angular-devkit/schematics    : 0.7.4
   @angular/cli                  : 6.1.4
   @ionic/angular-toolkit        : not installed

Cordova:

   cordova (Cordova CLI) : not installed
   Cordova Platforms     : android 7.1.4, ios 4.5.5
   Cordova Plugins       : not available

System:

   Android SDK Tools : 26.1.1 (C:\Users\Stationaer\AppData\Local\Android\Sdk)
   NodeJS            : v8.9.0 (C:\Program Files\nodejs\node.exe)
   npm               : 6.4.1
   OS                : Windows 10
```

Maybe I am missing something, but I'm experiencing a problem with Ionic caching views previously visited, preventing the components from being re-initialized by Angular. The components are pulled from some cache and rendered as whatever data existed previously.

---
Example:
User A is logged into the application and starts on the 'Home' page with info relevant to user A. User A logs out and navigates to 'Log In'.

User B logs in from the same application and navigates to 'Home' page. Ionic see's that Home was previously visited (by user A) and instead of instantiating 'Home', the view is pulled from cache and displays everything that User A was seing.

---
I noticed ion-router-outlet adds page transitioning when navigation, and this transition swaps to the left, when the navigation goes "forward" and swipe right when the navigation goes "backwards". This data seems to read from the same history-source that stores cached views.

TL;DR How to control Ionic router history (url tree) from caching the views preventing angular to re-initialize the components?

[StackOverflow question](https://stackoverflow.com/questions/53395620/ionic-4-angular-6-how-to-control-ionic-router-history-to-stop-cache-a-view-co)

**Comment by KevinKelchen** [comment]
In case it's helpful, I've described some potentially relevant behavior [in this comment](https://github.com/ionic-team/ionic/issues/16367#issuecomment-440734525) where Pages get re-used (not destroyed and re-initialized) from the "stack."

**Comment by manucorporat** [comment]
You probably want to use:
```
navController.navigateRoot()
```

or
```
<ion-button [routerLink]="page" [routerDirection]="root>
```
in order to clean all the pages in the stack.

**Comment by Moelgaard85** [comment]
@manucorporat Thank you for the answer. I implemented your solution and it seems to have some effect because the components constructor functions are now being fired.

But thats the only thing being fired. The ngOnInit is not being fired as normal initialization of Angular componetns. So it seems the components are being re-constructed, but not re-initialized still leaving me with components not being initialised. Any ideas?
